shared/src/iosMain/kotlin/com/karumi/gallery/domain/TimeProvider.kt

```diff
@@ -0,0 +1,36 @@
+package com.karumi.gallery.domain
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.ptr
+import platform.Foundation.NSUserDefaults
+import platform.posix.gettimeofday
+import platform.posix.timeval
+
+actual class TimeProvider {
+
+  companion object {
+    private const val TIME_ARG = "time_arg"
+    private const val THREE_MONTHS_IN_MILLIS = 259200
```

Comment:
```suggestion
    private val THREE_MONTHS_IN_MILLIS = 90.days.toLongMilliseconds()
```

Reply:
Now that the specific time is encoded into the value we can rename the variable to its actual meaning like `TIME_IN_MS_TO_EXPIRE_DATA` or something like that

Reply:
yes, I think either is fine, three months in millis is the number and time in ms to expire data is the purpose of that number.

But I forgot to use the `kotlin.time` the approach I used for the TTLCache -.-, good catch 💪